source/IncomingMessages.Domain/MessageParsers/JsonMessageParserBase.cs

```diff
@@ -93,69 +78,136 @@ protected override async Task<IncomingMarketMessageParserResult> ParseMessageAsy
 
     protected abstract IncomingMarketMessageParserResult CreateResult(MessageHeader header, IReadOnlyCollection<IIncomingMessageSeries> transactions);
 
-    private static string GetJsonDateStringWithoutQuotes(JsonElement element)
+    private static string? GetBusinessType(JsonElement element)
     {
-        return element.ToString().Trim('"');
+        return element.TryGetProperty("businessSector.type", out var property) ? property.GetProperty("value").GetString() : null;
     }
 
-    private static string? GetBusinessType(JsonElement element)
+    private List<IIncomingMessageSeries>? ValidateAndParseTransactions(
+        JsonDocument document,
+        JsonSchema schemaResult)
+    {
+        var headerElement = GetHeaderElement(document);
+        if (headerElement == null)
+            return null;
+
+        var transactionElements = document.RootElement.GetProperty(HeaderElementName).GetProperty(SeriesElementName);
+        var transactions = new List<IIncomingMessageSeries>(transactionElements.GetArrayLength());
+
+        foreach (var transactionElement in transactionElements.EnumerateArray())
+        {
+            // This validates a full document that contains only a single transaction.
+            // By isolating one transaction per validation, we reduce peak memory usage
+            // and allow memory to be reused between validations.
+            using var documentWithOneTransaction = GetDocumentWithOneSeriesElement(transactionElement, headerElement);
+            if (!IsValid(documentWithOneTransaction, schemaResult))
+                return null;
+
+            var header = ParseHeader(documentWithOneTransaction!);
+            transactions.Add(ParseTransaction(
+                documentWithOneTransaction!.RootElement.GetProperty(HeaderElementName).GetProperty(SeriesElementName).EnumerateArray().First(),
```

Comment:
We should be able to use `.Single()` here, since we know there is only one transaction in the temporary document.

```suggestion
                documentWithOneTransaction!.RootElement.GetProperty(HeaderElementName).GetProperty(SeriesElementName).EnumerateArray().Single(),
```